src/components/ingestion/AbstractIngestionEngine.js

```diff
@@ -1,66 +1,120 @@
 import { spawn } from 'child_process'
+import {
+	pipeline,
+	Writable,
+	Transform,
+} from 'stream'
 import { Payload } from '@tvkitchen/base-classes'
 import { dataTypes } from '@tvkitchen/base-constants'
 import {
 	AbstractInstantiationError,
 	NotImplementedError,
 } from '@tvkitchen/base-errors'
+import { TSDemuxer } from 'ts-demuxer'
+
 import logger from '%src/lib/logger'
 import kafka from '%src/lib/kafka'
+import { tsToSeconds } from '%src/lib/utils/mpegts'
 
 class AbstractIngestionEngine {
 	constructor() {
 		if (this.constructor === AbstractIngestionEngine) {
 			throw new AbstractInstantiationError(this.constructor.name)
 		}
 		this.producer = kafka.producer()
+		this.mpegtsDemuxer = new TSDemuxer(this.processDemuxedPacket)
+		this.mostRecentDemuxedPacket = null
 	}
 
 	/**
-	 * Ingests data into the TV Kitchen pipeline.
+	 * Ingests an mpegts payload into the TV Kitchen pipeline.
 	 *
-	 * @param  {Buffer} data The buffer of data to be ingested.
-	 * @return {Boolean}     Whether or not the buffer was successfully ingested.
+	 * @param  {Payload} data The payload to be ingested.
+	 * @return {Boolean}      Whether or not the payload was successfully ingested.
```

Comment:
I think I'm out of my CS depth here, because I am not following how we pass params into `ingestPayloadStream`, and I _think_ this `@return` is now wrong? Isn't it returning a Promise or something? Gah!

Okay so does the `write()` method of the `Writable` object receive the parameters passed to the `ingestPayloadStream` function? If so, I suppose we can rely on people understanding that (to the extent they understand Node streams better than I do), but write now this function is Greek to me and the JSDoc block doesn't seem to align with _reading_ the code, but maybe this does!

So, this is just a request to double-check the comments and the function and give a ✅ that it's in sync as you understand it, since I don't.

Reply:
I think this relates to what I was trying to say [here](https://github.com/tvkitchen/tv-kitchen/pull/47#discussion_r432083316).

Reply:
Jutin you are right to be confused, this was all wrong.

This is not a method, it is an instance variable.  It shouldn't be `ingestPayloadStream` (which is an imperative) but rather `payloadIngestionStream` (which is a noun).

And yes this pairs directly with Chris's excellent comment!

Thanks all.